mapie/control_risk/ltt.py

```diff
@@ -0,0 +1,131 @@
+
+    delta: float.
+        Probability of not controlling empirical risk.
+        Correspond to proportion of failure we don't
+        want to exceed.
+
+    Returns
+    -------
+    valid_index: List[List[Any]].
+        Contain the valid index that satisfy fwer control
+        for each alpha (length aren't the same for each alpha).
+
+    p_values: NDArray of shape (n_lambda, n_alpha).
+        Contains the values of p_value for different alpha.
+
+    References
+    ----------
+    [1] Angelopoulos, A. N., Bates, S., Candès, E. J., Jordan,
+    M. I., & Lei, L. (2021). Learn then test:
+    "Calibrating predictive algorithms to achieve risk control".
+    """
+    if delta is None:
+        raise ValueError(
+            "Invalid delta: delta cannot be None while"
+            + " controlling precision with LTT. "
+        )
+    p_values = compute_hoeffdding_bentkus_p_value(r_hat, n_obs, alpha_np)
+    N = len(p_values)
+    valid_index = []
+    for i in range(len(alpha_np)):
+        l_index = np.where(p_values[:, i] <= delta/N)[0].tolist()
+        valid_index.append(l_index)
+    return valid_index, p_values
+
+
+def find_lambda_control_star(
+    r_hat: NDArray,
+    valid_index: List[List[Any]],
+    lambdas: NDArray
+) -> Tuple[ArrayLike, ArrayLike]:
+    """
+    Return the lambda that give the maximum precision with a control
+    guarantee of level delta.
+    Note: When a list inf valid_index is empty, we can assume that
+    lambda is equal to 1, as no lambda can verify the fwer control.
+    And so to make our statement clear we also assume that the risk is
+    high.
+
+    Parameters
+    ----------
+    r_hat: NDArray of shape (n_lambdas, n_alpha)
+        Empirical risk with respect
+        to the lambdas and to the alphas.
+        Here, lambdas are thresholds that impact decision-making
+        and therefore the empirical risk.
+        Alphas are levels of empirical risk control such that
+        the empirical risk is less than alpha.
+
+    valid_index: List[List[Any]].
+        Contain the valid index that satisfy fwer control
+        for each alpha (length aren't the same for each alpha).
+
+    lambdas: NDArray of shape (n_lambda, )
+        Discretize parameters use for ltt procedure.
+
+    Returns
+    -------
+    l_lambda_star: ArrayLike of shape (n_alpha, ).
+        The lambda that give the highest precision
+        for a given alpha.
+
+    r_star: ArrayLike of shape (n_alpha, ).
+        The value of lowest risk for a given alpha.
+    """
+    if [] in valid_index:
+        warnings.warn(
+            """
+            Warning: At least one sequence is empty!
+            """
+        )
+    l_lambda_star = []  # type: List[Any]
+    l_r_star = []  # type: List[Any]
+    for i in range(len(valid_index)):
+        if len(valid_index[i]) == 0:
+            l_lambda_star.append(1)
+            l_r_star.append(1)
+        else:
+            idx = np.argmin(r_hat[valid_index[i]])
```

Comment:
Here, you take the index of the lambdas (among all valid lambas) which minimizes your risk. 
Let say you want to find the optimal lambda for two alphas = [.2, .5]. In the case where both values of alpha ahve valid lambdas, then for both values of alphas you will get the same lambda_star (as for both alphas, the same minimal risk will be achieved). I think maybe it would be better to always give the lower possible value of alpha because it will both guarantee your precision, without decreasing to much your recall. This change can be made by replacing:
`idx = np.argmin(r_hat[valid_index[i]])` by `idx = np.argmin(valid_index[i])`

Reply:
@thibaultcordier what do you think about this ?

Reply:
I agree with your argument. The property "it will both guarantee your precision, without decreasing to much your recall" is pretinent. In this case, it should be included in the docstring/documentation.

Reply:
I agree too, but what if user want lambda such that he has maximum precision? How can i let him know the procedure?

Reply:
What do you think if I had a third output that will contain the lambda that give the highest precision?

Reply:
I agree, let output 3 values for this function, in the main mapie code, we should use as threshold only the one which correspond to the minimal precision, and we give to the user to threshold which maximizes the precision as a class attribute

Reply:
I don't quite agree with this option. Why not add a parameter that defines the behaviour we want?

Reply:
Done

Reply:
@thibaultcordier because I don't really see a use case where it would be useful, and I think it's better to not multiply the number of arguments, this solution ables us to keep the class a little simpler

Reply:
I think I misunderstood the proposal in this case. For the main behavior, I agree with you. 
If we want to choose between optimizing the risk or minimizing the lambdas, this feature can be seen later (in a future PR). That's what I gather from your discussions and the changes underway.